eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -45,8 +45,19 @@ jobs:
 
       - template: ../steps/common.yml
 
+      - script: |
+          node eng/tools/generate-dev-builds.js "$(Build.BuildNumber)"
+        condition: eq(variables['DevBuildRun'],'true')
+        displayName: "Update package versions for dev build"
+
+      - script: |
+          node common/scripts/install-run-rush.js update
+        condition: eq(variables['DevBuildRun'],'true')
+        displayName: "Install dependencies with updates"
+
       - script: |
           node common/scripts/install-run-rush.js install
+        condition: ne(variables['DevBuildRun'],'true')
```

Comment:
Do we need to avoid the install after the update?

Reply:
We need to do `update` if we updated the version numbers, and `install` otherwise. We don't want to do both. Install after update shouldn't do anything but it will take extra time and is unnecessary.

Reply:
Should i change the order of the steps to this = 1> rush install  2> generate-dev-builds 3> rush update

Reply:
I don't think we want to confuse things by doing an install of the packages before you update to the dev versions.

Reply:
Actually sorry for the confusion, I already have the conditions set to - eq(variables['DevBuildRun'],'true') for rush update and ne(variables['DevBuildRun'],'true') for rush install